### DISCUSSION TITLE
Lineweight

### DISCUSSION THREAD

**Discussion by mindreaderlupo** [tarikjabiri/dxf#138] [Q&A]
Hi,

I am trying to rebuild existing Autocad files with this lib to automate drawing generation.

In the documentation, I got the point how to create Layers and Linetypes. I am currently struggeling to attatch Lineweights to an Entity, Block or (this would be the best case) a Layer.
I need Lineweight for 2 reasons: 1. Printing, 2. create bold text blocks

Could someone help me with a small example?

Thanks a lot and best regards
Felix

**Comment by tarikjabiri** [comment]
Hi,

unfortunately in the current version you cant customize that.

But in the next version this will be doable, still the `next` branch in development. 

example using the new api:
```ts
const writer = new Writer();
const modelSpace = writer.document.modelSpace;

// Display the Lineweight
writer.document.addVariable("$LWDISPLAY").add(290, 1);

const green = writer.document.tables.addLayer({
  name: "Green",
  colorNumber: Colors.Green,
});

modelSpace.addLine({
  start: point(),
  end: point(100, 100),
  lineWeight: 200, // Set the Lineweight. Should be the value * 100, not sure why!
  layerName: green.name,
});
```
Stay tuned, the next version will be 3.0.0.

Regards.

**Reply by mindreaderlupo** [reply]
Thanks! Looks like some breaking changes but totally worth it :)

**Reply by tarikjabiri** [reply]
Yeah, I am trying to fix some problems in the previous implementation.

Just need more time until its ready to publish.

**Comment by tarikjabiri** [comment]
Hi

I published an alpha version.

there is support for [lineweights](https://dxf-ikomiqhon-dxfjs.vercel.app/guides/lineweights.html).

[https://www.npmjs.com/package/@tarikjabiri/dxf/v/3.0.0-alpha.1](https://www.npmjs.com/package/@tarikjabiri/dxf/v/3.0.0-alpha.1)
